source/extensions/filters/http/buffer/buffer_filter.cc

```diff
@@ -95,6 +95,15 @@ Http::FilterDataStatus BufferFilter::decodeData(Buffer::Instance& data, bool end
 }
 
 Http::FilterTrailersStatus BufferFilter::decodeTrailers(Http::HeaderMap&) {
+  // request_headers_ is initialized iff plugin is enabled.
+  if (request_headers_ != nullptr && request_headers_->ContentLength() == nullptr) {
+    ASSERT(!settings_->disabled());
+    if (Runtime::runtimeFeatureEnabled(
+            "envoy.reloadable_features.buffer_filter_populate_content_length")) {
+      request_headers_->insertContentLength().value(content_length_);
+    }
+  }
+
```

Comment:
Can this code be moved to a shared function, please?
It's non-trivial enough to care about the code duplication.